pkg/cnab/provider/driver_test.go

```diff
@@ -52,6 +53,28 @@ func TestNewDriver_Docker(t *testing.T) {
 		assert.EqualError(t, err, "allow-docker-host-access was specified but could not detect a local docker daemon running by checking for /var/run/docker.sock")
 	})
 
-	// TODO: add tests that check the Docker configuration options based on the provided required extension config.
-	// Requires changes in cnab-go (export of configurationOptions field on Driver struct in docker.go)
+	t.Run("docker with host access, privileged true", func(t *testing.T) {
+		d := NewTestRuntime(t)
+		d.Extensions[extensions.DockerExtensionKey] = extensions.Docker{
+			Privileged: true,
+		}
+		d.FileSystem.Create("/var/run/docker.sock")
+		args := ActionArguments{
+			AllowDockerHostAccess: true,
+		}
+
+		driver, err := d.newDriver(DriverNameDocker, "myclaim", args)
+		require.NoError(t, err)
+		assert.IsType(t, driver, &docker.Driver{})
+
+		dockerish, ok := driver.(*docker.Driver)
+		assert.True(t, ok)
+
+		err = dockerish.ApplyConfigurationOptions()
+		assert.NoError(t, err)
+
+		containerHostCfg, err := dockerish.GetContainerHostConfig()
+		require.NoError(t, err)
+		require.Equal(t, true, containerHostCfg.Privileged)
```

Comment:
This is perfect. Thanks for your work getting the plumbing into cnab-go to support it.

Can you add a converse test to verify that when we don't have privileged that it's not set on the driver? Just to check for us hard coding it accidentally like it was originally.